### PR TITLE
Feature/application json

### PR DESCRIPTION
**Pull request by paed01**
auto-parse content-type application/json

### CODE REVIEW

File: test/http-test.js [hunk 1]
```diff
@@ -416,6 +416,26 @@ describe("httptest", () => {
...
+
+    it("content-type application/json with text body throws", async () => {
+      const resp = await request(app)
+        .post("/api")
+        .send({ a: "b" })
+        .expect("content-type", "application/json; charset=utf-8")
+        .expect(201);
+
+      expect(resp.body).to.deep.equal({ a: "b" });
+    });
```
Comment (jonaswalden) [inline test/http-test.js]: does description actually match assertions? seems identical to previous test

File: index.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ const got = require("got");
 const assert = require("assert");
 const { CookieJar } = require("cookiejar");
 
+const isJson = /application\/json/i;
```
Comment (jonaswalden) [inline index.js]: prefix `is` makes it sound like the result of a type evaluation.
maybe `jsonContentTypePattern`. 
🤷

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,7 +8,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [2.0.0] - 2022-08-22
+
+### Breaking
```
Comment (jonaswalden) [inline CHANGELOG.md]: Keep a Changelog (which this changelog claims to be formatted after up top) uses "changed"

```suggestion
### Changed
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,7 +8,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [2.0.0] - 2022-08-22
```
Comment (jonaswalden) [inline CHANGELOG.md]: release not out yet. date should be changed before publish.
better to keep changes in the unreleased section until actually releasing. easier to maintain document in parallell PRs etc.

File: test/http-test.js (hunk 1 above)
Comment (paed01) [reply test/http-test.js]: No it doesn't, sorry.

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,8 +7,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
-
-## [1.1.0] - 2022-04-13
+### Changed
+- Parse and return response body as JSON if content type matches application/json
+#
+# [1.1.0] - 2022-04-13
```
Comment (jonaswalden) [inline CHANGELOG.md]:
```suggestion

### Changed
- Parse and return response body as JSON if content type matches application/json

## [1.1.0] - 2022-04-13
```

File: test/http-test.js [hunk 6]
```diff
@@ -416,6 +416,26 @@ describe("httptest", () => {
...
+
+    it("content-type application/json returns body as JSON", async () => {
+      const resp = await request(app)
+        .post("/api")
+        .send({ a: "b" })
+        .expect("content-type", "application/json; charset=utf-8")
+        .expect(201);
+
+      expect(resp.body).to.deep.equal({ a: "b" });
+    });
```
Comment (jonaswalden) [inline test/http-test.js]: can't see the difference between this test and the one above other than json being uppercased in this description. am I missing something?

File: CHANGELOG.md [hunk 7]
```diff
@@ -7,8 +7,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- Parse and return response body as JSON if content type matches application/json
 
-## [1.1.0] - 2022-04-13
+# [1.1.0] - 2022-04-13
```
Comment (jonaswalden) [inline CHANGELOG.md]: xD

```suggestion
## [1.1.0] - 2022-04-13
```

**Review by jonaswalden — APPROVED** [review]

**Pull request merged**